### PR TITLE
Update Basic Cluster to align with CHIP spec

### PR DESCRIPTION
**Pull request by kpschoedel**
#### Problem

The currently implemented Basic Cluster is pulled unmodified from
the ZCL spec and does not align with the CHIP defined Basic Cluster.

#### Summary of Changes

- update `general.xml` for Basic cluster
- rename a field in `ta.xml` and `application-basic-cluster.xml` to avoid conflicts
- update `.zap` files
- regenerate artifacts

issue #2877 Update Basic Cluster to align with CHIP spec

### CODE REVIEW

**Comment by vivien-apple** [comment]
@kpschoedel  I guess I may be missing the right pointer to the spec. The spec I have handy does not have the same attributes/commands. Mind sharing the spec source you used ?

**Comment by kpschoedel** [comment]
> @kpschoedel I guess I may be missing the right pointer to the spec. The spec I have handy does not have the same attributes/commands. Mind sharing the spec source you used ?

Yes, I had the same confusion. The CHIP Basic Cluster is this one: https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/service_device_management/BasicCluster.adoc and _not_ src/app_clusters/Basic.adoc ([which is from ZCL](https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/1957)).

**Comment by kpschoedel** [comment]
Also, this PR incorporates https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/923 which isn't in the text yet.

File: src/app/zap-templates/zcl/data-model/silabs/general.xml
Comment (vivien-apple) [inline src/app/zap-templates/zcl/data-model/silabs/general.xml:183]: MODEL -> PRODUCT ?

File: src/app/zap-templates/zcl/data-model/silabs/general.xml
Comment (vivien-apple) [inline src/app/zap-templates/zcl/data-model/silabs/general.xml:187]: REVISION -> VERSION ?

File: src/app/zap-templates/zcl/data-model/silabs/general.xml
Comment (vivien-apple) [inline src/app/zap-templates/zcl/data-model/silabs/general.xml:188]: REVISION -> VERSION ?

File: src/app/zap-templates/zcl/data-model/silabs/general.xml
Comment (vivien-apple) [inline src/app/zap-templates/zcl/data-model/silabs/general.xml:189]: I guess you're out of luck. The spec attributes has changed since your patch. You will need to rebased, please re-request review once it is done.

File: src/app/zap-templates/zcl/data-model/silabs/general.xml
Comment (vivien-apple) [inline src/app/zap-templates/zcl/data-model/silabs/general.xml:189]: INT16U -> INT32U

**Review by vivien-apple — APPROVED** [review]
Looks good. Approving assuming the `SoftwareVersion` type will be changed before merging.

File: src/app/zap-templates/zcl/data-model/silabs/general.xml
Comment (kpschoedel) [reply src/app/zap-templates/zcl/data-model/silabs/general.xml]: Thanks for catching that.

I still need to update the cluster ID and do all the .zap conversions.

File: src/app/zap-templates/zcl/data-model/silabs/general.xml
Comment (kpschoedel) [inline src/app/zap-templates/zcl/data-model/silabs/general.xml:179]: dummy comment

File: examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h
Comment (bzbarsky-apple) [inline examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h:45]: Any idea why we are not getting a newline before the start of the new attribute?  Makes this pretty hard to read....

@vivien-apple

**Review by bzbarsky-apple — APPROVED** [review]

**Review by jelderton — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Comment by woody-apple** [comment]
@kpschoedel Looks ready for merge, short of the conflicts - can you resolve conflicts here?

File: examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h
Comment (vivien-apple) [reply examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h:45]: @bzbarsky-apple  https://github.com/project-chip/zap/pull/117

**Comment by erjiaqing** [comment]
Seems this PR breaks ZAP template generation:

https://github.com/project-chip/connectedhomeip/runs/2033639974 -> Success
https://github.com/project-chip/connectedhomeip/runs/2033644279 -> Failed

**Comment by yuzibo** [comment]
I just wonder where is CHIP spec that you are talking about? It looks like all CHIP spec's urls are broken.

**Comment by bzbarsky-apple** [comment]
> It looks like all CHIP spec's urls are broken.

Github shows a 404 if you don't have permissions to see the relevant page.  It sounds like you don't have permissions to the CHIP spec repo...

**Comment by kpschoedel** [comment]
> Can I get this permissions to access CHIP spec? In other word, How can I obtain the permissions?

I think you have to join the Zigbee Working Group, starting with the form on [connectedhomeip.com](https://www.connectedhomeip.com/). I believe the spec will be public as soon as it's ready (they don't want people relying on the draft that will still be changed) but none of us developers here on github can speak for the organization.

**Comment by clapre** [comment]
@yuzibo Kevin is (mostly) right. While the Connected Home over IP project is using Open Source to develop the implementation, we are working on the specification within the Alliance. I would be happy to meet with you to answer any questions on the project, the specification, or joining the Alliance you may have. A quicker way is to reach out to help@zigbee.org and mention that you want to set up a call with Chris and we can set that up. Thanks for you interest!

**Comment by yuzibo** [comment]
Ok，I understand the work-flow for CHIP basically now. Thank you for all.

**Pull request merged**
